### DISCUSSION TITLE
spawning PopupInput Dialogue

### DISCUSSION THREAD

**Discussion by lukas-runge** [hossimo/GMA3Plugins#22] [Q&A] [answered]
Hey @hossimo and everyone else,

I am trying to spawn a **PopupInput** in the current MA3 Version (v1.4.0.2) but cant get it working.
I am trying the following example from the [wiki](https://github.com/hossimo/GMA3Plugins/wiki/PopupInput):

**`lua 'local r = PopupInput("TITLE", display_handle, {{"str", "String", nil}, {"int","Number", 123}}, "123", 100, 200)'`**

**What am I doing wrong?**

Kind regards
@lukas-runge

**Comment by hossimo** [comment, marked as answer]
The issue is because the second argument of [PopupInput()](https://github.com/hossimo/GMA3Plugins/wiki/PopupInput) is looking for a `display_handle`.

In the system monitor you should notice an error like this:
![image](https://user-images.githubusercontent.com/1986602/102420128-662df480-3fcf-11eb-9749-9119f3639b31.png)

It's not 1000% clear but what it is saying is the second argument is the wrong type. it's looking for a LUA lightuserdata object but instead it got something else, `display_handle`, which in this context is likely `nil`. In a proper LUA file you will notice that the main function of your program has a `display_handle` argument:

``` lua
local function Main(display_handle,argument)
    local r = PopupInput("TITLE", display_handle, {{"str", "String", nil}, {"int","Number", 123}}, "123", 100, 200)
end
```

In this case `display_handle` passed in via the system to the main function of your program.

If `PopupInput()` worked with a `nil` value it was probably a bug that was fixed.

### Two ways around this:

## 1) Get the display Handle

I had noticed some functions in the API like `HandleToInt()` and `IntToHandle()` that you could might be able to use to get the int value of the handle and use it on the command line. I have it a try in a lua file:

``` lua
local function Main(display_handle,argument)
  Echo(HandleToInt(display_handle)) -- returned 838861405
end
return Main
```

Then I used this number on the command line:

``` lua
lua 'local r = PopupInput("TITLE", IntToHandle(838861405), {{"str", "String", nil}, {"int","Number", 123}}, "123", 100, 200)'
```

And it worked, HOWEVER this number probably changes for each session, so you would need some way to find this number.

## 2) Use [GetDisplayByIndex()](https://github.com/hossimo/GMA3Plugins/wiki/GetDisplayByIndex)

`lua 'local r = PopupInput("TITLE",GetDisplayByIndex(1), {{"str", "String", nil}, {"int","Number", 123}}, "123", 100, 200)'`

Some guessing on my part but I believe `GetDisplayByIndex()` returns a display handle for a 1 based display index number, so `GetDisplayByIndex(1)` returns the Display handle for display 1, again guessing on my part. you may want to ask on the forums to see if anyone answers.

Hope this helps.

**Reply by lukas-runge** [reply]
Thank you very much @hossimo! The missing `display_handle` made it work! 😍

Kind regards
@lukas-runge
